pages/1.13/administering-clusters/locate-public-agent/index.md

```diff
@@ -6,23 +6,193 @@ menuWeight: 3
+    10.0.1.112     10.0.1.112                  ecb5e39c-2d3e-4eea-8c07-af0c4e9e8443-S0  agent             aws/us-west-2  aws/us-west-2d  
+    master.mesos.  10.0.6.157  34.222.201.246  ecb5e39c-2d3e-4eea-8c07-af0c4e9e8443     master (leader)   aws/us-west-2  aws/us-west-2d  
+    master.mesos.  10.0.7.169  34.223.44.83    N/A                                      master (standby)  aws/us-west-2  N/A             
+    master.mesos.  10.0.7.38   34.222.181.165  N/A                                      master (standby)  aws/us-west-2  N/A             
+    ```
+
+    From this output, you can identify the specific public-facing IP addresses to use for both public and private agent nodes and for master nodes.
+
+    In this example, there is one public IP address for the public agent:
+    - 34.223.48.55
+    
+    There are three public IP addresses for the master nodes:
+    - 34.222.201.246 (leader)
+    - 34.223.44.83 (standby)
+    - 34.222.181.165 (standby)
+
+    There are no public-facing IP addresses available for the private agent node. 
+
+<a name="public-ip-api">
+
+# Finding a public IP address using an API call
+The DC/OS application programming interface (API) provides the underlying funtionality that you can access through the DC/OS web-based administrative console and command-line interface (CLI). In most cases, therefore, you only use the API directly if you are integrating the functionality with a custom program or automation script. However, you can retrieve the public IP addresses for public agents directly through calls to the DC/OS application programming interface (API) for networking, if needed.
+
+To find public IP addresses by using the DC/OS API:
+1. Identify the URL for a DC/OS cluster running on a public cloud instance.
+
+1. Use the following REST API `net/v1/nodes` endpoint to find the public IP of the public agents:
+
+    ```bash
+    <cluster-url>/net/v1/nodes
+    ```
+
+    For example, you might issue a call with a cluster URL similar to the following to specify where the DC/OS cluster is running:
+    ```bash
+    http://luxi-sf7-elasticl-u70m3un6kcab-1100943753.us-west-2.elb.amazonaws.com/net/v1/nodes
+    ```
+
+1. Review the command output.
+
+    This API call returns information similar to the following in an unfiltered format:
+
+    ```bash
+    [{"updated":"2019-01-07T22:22:22.171Z","public_ips":["34.212.37.79"],"private_ip":"10.0.6.210","hostname":"ip-10-0-6-210"},{"updated":"2019-01-07T22:22:22.119Z","public_ips":["52.25.254.97"],"private_ip":"10.0.6.181","hostname":"ip-10-0-6-181"},{"updated":"2019-01-07T22:21:09.585Z","public_ips":["54.218.23.75"],"private_ip":"10.0.6.148","hostname":"ip-10-0-6-148"},{"updated":"2019-01-07T22:22:28.582Z","public_ips":[],"private_ip":"10.0.1.139","hostname":"ip-10-0-1-139"},{"updated":"2019-01-07T22:22:28.649Z","public_ips":[],"private_ip":"10.0.0.138","hostname":"ip-10-0-0-138"}]
+    ```
+
+    In this example, there are three IP addresses for the public agents:
```

Comment:
One of these IPs is on the master node (assuming this was from a 1 master and 2 public, 2 private agent setup). Not sure really how to phrase this correctly, but technically not all the ips are on public agents.